### PR TITLE
Adding capability to stop an execution forcefully

### PR DESCRIPTION
**Pull request by supunkamburugamuve**
This can help to stop the task examples to stop

### CODE REVIEW

File: twister2/executor/src/java/edu/iu/dsc/tws/executor/threading/StreamingSharingExecutor.java [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+          INodeInstance nodeInstance = tasks.poll();
+          if (nodeInstance != null) {
+            nodeInstance.execute();
+            tasks.offer(nodeInstance);
+          } else {
+            LOG.log(Level.INFO, "Thread existing as more threads than tasks "
+                + "are been assigned");
+            break;
+          }
+        } catch (Throwable t) {
```
Comment (chathurawidanage) [inline twister2/executor/src/java/edu/iu/dsc/tws/executor/threading/StreamingSharingExecutor.java]: Is there a possibility of leaking INodeInstances here?

Comment (supunkamburugamuve) [reply twister2/executor/src/java/edu/iu/dsc/tws/executor/threading/StreamingSharingExecutor.java]: I think not by looking at it. But not 100% sure.

Comment (supunkamburugamuve) [reply twister2/executor/src/java/edu/iu/dsc/tws/executor/threading/StreamingSharingExecutor.java]: Fixed this one

File: twister2/executor/src/java/edu/iu/dsc/tws/executor/threading/BatchSharingExecutor.java [hunk 2]
```diff
@@ -90,34 +103,27 @@ public boolean runExecution() {
...
 
+    scheduleExecution(nodes);
+    return new BatchExecution(nodes);
   }
 
   protected class BatchWorker implements Runnable {
     @Override
     public void run() {
-      while (true) {
+      while (notStopped) {
```
Comment (chathurawidanage) [inline twister2/executor/src/java/edu/iu/dsc/tws/executor/threading/BatchSharingExecutor.java]: This block has the same issue...

Comment (supunkamburugamuve) [reply twister2/executor/src/java/edu/iu/dsc/tws/executor/threading/BatchSharingExecutor.java]: This is fixed now

**Pull request merged**